Userland/Libraries/LibJS/AST.h

```diff
@@ -137,6 +139,25 @@ class ExpressionStatement final : public Statement {
     NonnullRefPtr<Expression> m_expression;
 };
 
+template<typename... Args>
+class IteratorOrVoidFunction : public Function<IterationDecision(Args...)> {
+public:
+    template<typename CallableType>
+    IteratorOrVoidFunction(CallableType&& callable) requires(VoidFunction<CallableType, Args...>)
+        : Function<IterationDecision(Args...)>([callable = move(callable)](Args... args) {
+            callable(forward<Args>(args)...);
+            return IterationDecision::Continue;
+        })
+    {
+    }
+
+    template<typename CallableType>
+    IteratorOrVoidFunction(CallableType&& callable) requires(IteratorFunction<CallableType, Args...>)
+        : Function<IterationDecision(Args...)>(move(callable))
```

Comment:
`callable` is a forwarding reference, use `forward()` instead of `move()`.
Also above.

Reply:
Yeah it was complaining about `forward` when I used that. But now it works so :shrug:. Thanks!